src/Testcontainers.Keycloak/KeycloakBuilder.cs

```diff
@@ -0,0 +1,124 @@
+﻿namespace Testcontainers.Keycloak;
+
+/// <inheritdoc />
+[PublicAPI]
+public sealed class KeycloakBuilder : ContainerBuilder<KeycloakBuilder, KeycloakContainer, KeycloakConfiguration>
+{
+	public const string KeycloakImage = "quay.io/keycloak/keycloak:21.1.1";
+	public const ushort KeycloakPort = 8080;
+	public const string DefaultUsername = "admin";
+	public const string DefaultPassword = "admin";
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="KeycloakBuilder"/> class.
+	/// </summary>
+	public KeycloakBuilder()
+		: this(new())
+	{
+		DockerResourceConfiguration = Init().DockerResourceConfiguration;
+	}
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="KeycloakBuilder"/> class.
+	/// </summary>
+	/// <param name="resourceConfiguration">The Docker resource configuration.</param>
+	private KeycloakBuilder(KeycloakConfiguration resourceConfiguration)
+		: base(resourceConfiguration)
+	{
+		DockerResourceConfiguration = resourceConfiguration;
+	}
+
+	/// <inheritdoc />
+	protected override KeycloakConfiguration DockerResourceConfiguration { get; }
+
+	/// <summary>
+	/// Sets the Keycloak admin username.
+	/// </summary>
+	/// <param name="username">The Keycloak admin username.</param>
+	/// <returns>A configured instance of <see cref="KeycloakBuilder"/>.</returns>
+	public KeycloakBuilder WithUsername(string username)
+	{
+		return Merge(DockerResourceConfiguration, new(username: username))
+			.WithEnvironment("KEYCLOAK_ADMIN", username)
+			.WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(KeycloakPort).AddCustomWaitStrategy(new WaitUntil(username)));
+	}
+
+	/// <summary>
+	/// Sets the Keycloak admin password.
+	/// </summary>
+	/// <param name="password">The Keycloak admin password.</param>
+	/// <returns>A configured instance of <see cref="KeycloakBuilder"/>.</returns>
+	public KeycloakBuilder WithPassword(string password)
+	{
+		return Merge(DockerResourceConfiguration, new(password: password))
+			.WithEnvironment("KEYCLOAK_ADMIN_PASSWORD", password);
+	}
+
+	/// <inheritdoc />
+	public override KeycloakContainer Build()
+	{
+		Validate();
+		return new(DockerResourceConfiguration, TestcontainersSettings.Logger);
+	}
+
+	/// <inheritdoc />
+	protected override KeycloakBuilder Init()
+	{
+		return base.Init()
+			.WithImage(KeycloakImage)
+			.WithCommand("start-dev")
+			.WithPortBinding(KeycloakPort, true)
+			.WithUsername(DefaultUsername)
+			.WithPassword(DefaultPassword);
```

Comment:
```suggestion
    return base.Init()
      .WithImage(KeycloakImage)
      .WithCommand("start-dev")
      .WithCommand("--health-enabled", "true")
      .WithPortBinding(KeycloakPort, true)
      .WithUsername(DefaultUsername)
      .WithPassword(DefaultPassword)
      .WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request =>
        request.ForPath("/health/ready").ForPort(KeycloakPort)));
```

What about using Keycloak's [HEALTHCHECK](https://www.keycloak.org/server/health) to determine readiness? Then we can remove it in `WithUsername(string)`.

Reply:
Yeah, that looks better, forgot to check the documentation. Fixed and rebased latest changes.